src/logic/hooks/useAsync.ts

```diff
@@ -3,32 +3,41 @@ import { useEffect, useState } from 'react'
 type AsyncResult<T> = {
   error: Error | undefined
   result: T | undefined
+  isLoading: boolean
 }
 
 const useAsync = <T>(asyncCall: () => Promise<T>): AsyncResult<T> => {
-  const [asyncVal, setAsyncVal] = useState<T>()
-  const [err, setErr] = useState<Error>()
+  const [result, setResult] = useState<T>()
+  const [error, setError] = useState<Error>()
+  const [isLoading, setIsLoading] = useState<boolean>(false)
 
   useEffect(() => {
     let isCurrent = true
 
-    setAsyncVal(undefined)
-    setErr(undefined)
+    setResult(undefined)
+    setError(undefined)
+    setIsLoading(true)
 
     asyncCall()
       .then((val: T) => {
-        if (isCurrent) setAsyncVal(val)
+        if (isCurrent) {
+          setResult(val)
+          setIsLoading(false)
+        }
       })
       .catch((error) => {
```

Comment:
You should rename this to `err`, otherwise it shadows the state variable of the same name.